ghcide/session-loader/Development/IDE/Session.hs

```diff
@@ -98,31 +98,38 @@ data SessionLoadingOptions = SessionLoadingOptions
   --   return the path for storing generated GHC artifacts,
   --   or 'Nothing' to respect the cradle setting
   , getCacheDirs :: String -> [String] -> IO CacheDirs
+  -- | Return the GHC lib dir to use for the 'unsafeGlobalDynFlags'
+  , getInitialGhcLibDir :: IO (Maybe LibDir)
   }
 
 defaultLoadingOptions :: SessionLoadingOptions
 defaultLoadingOptions = SessionLoadingOptions
     {findCradle = HieBios.findCradle
     ,loadCradle = HieBios.loadCradle
     ,getCacheDirs = getCacheDirsDefault
+    ,getInitialGhcLibDir = getInitialGhcLibDirDefault
     }
 
--- | Sets `unsafeGlobalDynFlags` on using the hie-bios cradle and returns the GHC libdir
-setInitialDynFlags :: IO (Maybe LibDir)
-setInitialDynFlags = do
+getInitialGhcLibDirDefault :: IO (Maybe LibDir)
+getInitialGhcLibDirDefault = do
   dir <- IO.getCurrentDirectory
   hieYaml <- runMaybeT $ yamlConfig dir
   cradle <- maybe (loadImplicitHieCradle $ addTrailingPathSeparator dir) HieBios.loadCradle hieYaml
```

Comment:
I would have made this line use `SessionLoadingOptions`, but you know your usecase the best.

Reply:
I considered that, would need to change the signature of `loadCradle` to take maybe a file path, which ends up being equivalent right?

Reply:
There is an identical line below in the other invocation of `loadCradle`, in `consultCradle`.